packages/fuchsia_ctl/lib/src/package_server.dart

```diff
@@ -121,4 +121,15 @@ class PackageServer {
     return OperationResult.error(
         'The "pm" executable exited with non-zero exit code.');
   }
+
+  /// Returns the source URL for the given local address.
+  ///
+  /// Calling this before calling [serveRepo] will result in a [StateError].
+  String sourceUrl(String localIp) {
+    if (localIp.contains(':')) {
```

Comment:
Did we validated this fixes the problem locally?

Reply:
Nope. And actually, I think it's wrong...

Reply:
Maybe try https://api.dartlang.org/stable/2.6.1/dart-io/InternetAddress-class.html and get the type from it?

Reply:
I tried that but InternetAddress fails to parse IPv6 link local addresses. Additionally we need the link local address from the perspective of the Fuchsia device, not the host. I'm working on an update to this PR that actually does that.

Reply:
When I ran locally " amberctl add_src" I just added the ipv6 + interface name from the host which is connected to the devices network and it worked. The device was able to access the amber server.

Devices in the lab are connected to a similar network setup " a nuc(host) with two interfaces one for internet access/lab network access and another one for internal nat network. The device is connected to the nat network. I think eno2 should work

Reply:
It may work but I think it's incorrect. With the second commit in this PR I've made the logic match that in `fx add-update-source`.